common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala

```diff
@@ -0,0 +1,93 @@
+package com.gu.mediaservice.lib.usage
+
+import java.net.URI
+
+import com.amazonaws.services.dynamodbv2.document.Item
+import com.gu.mediaservice.model.usage._
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object ItemToMediaUsage {
+
+  def transform(item: Item) = {
+    MediaUsage(
+      UsageId(item.getString("usage_id")),
+      item.getString("grouping"),
+      item.getString("media_id"),
+      UsageType(item.getString("usage_type")),
+      item.getString("media_type"),
+      UsageStatus(item.getString("usage_status")),
+      Option(item.getMap[Any]("print_metadata"))
+        .map(_.asScala.toMap).flatMap(buildPrint),
+      Option(item.getMap[Any]("digital_metadata"))
+        .map(_.asScala.toMap).flatMap(buildDigital),
+      Option(item.getMap[Any]("syndication_metadata"))
+        .map(_.asScala.toMap).flatMap(buildSyndication),
+      Option(item.getMap[Any]("front_metadata"))
+        .map(_.asScala.toMap).flatMap(buildFront),
+      new DateTime(item.getLong("last_modified")),
+      Try {
+        item.getLong("date_added")
+      }.toOption.map(new DateTime(_)),
+      Try {
+        item.getLong("date_removed")
+      }.toOption.map(new DateTime(_))
+    )
+  }
+
+  private def buildFront(metadataMap: Map[String, Any]): Option[FrontUsageMetadata] = {
+    Try {
+      FrontUsageMetadata(
+        metadataMap("addedBy").asInstanceOf[String],
+        metadataMap("front").asInstanceOf[String]
+      )
+    }.toOption
+  }
+
+  private def buildSyndication(metadataMap: Map[String, Any]): Option[SyndicationUsageMetadata] = {
+    Try {
+      SyndicationUsageMetadata(
+        metadataMap("partnerName").asInstanceOf[String]
+      )
+    }.toOption
+  }
+
+  private def buildDigital(metadataMap: Map[String, Any]): Option[DigitalUsageMetadata] = {
+    Try {
+      DigitalUsageMetadata(
+        URI.create(metadataMap("webUrl").asInstanceOf[String]),
+        metadataMap("webTitle").asInstanceOf[String],
+        metadataMap("sectionId").asInstanceOf[String],
+        metadataMap.get("composerUrl").map(x => URI.create(x.asInstanceOf[String]))
+      )
+    }.toOption
+  }
+
+  private def buildPrint(metadataMap: Map[String, Any]): Option[PrintUsageMetadata] = {
+    type JStringNumMap = java.util.LinkedHashMap[String, java.math.BigDecimal]
+    Try {
+      PrintUsageMetadata(
+        sectionName = metadataMap.apply("sectionName").asInstanceOf[String],
+        issueDate = metadataMap.get("issueDate").map(_.asInstanceOf[String])
+          .map(ISODateTimeFormat.dateTimeParser().parseDateTime).get,
```

Comment:
is the `.get` safe here?

Reply:
Ah... suppose it's surrounded by a Try.

Reply:
I haven't put much attention to details as most of it is moving code around as building blocks not in details what the code does.

I did not introduced some new implementations, i mainly change the structure

Reply:
that function was present before as MediaUsageOps.build: https://github.com/guardian/grid/blob/master/usage/app/model/MediaUsage.scala#L44